### PR TITLE
Auto check the current year in boilerplate

### PR DESCRIPTION
**Pull request by fisherxu**
**What this PR does / why we need it**:
Every new year we need to add the current year to boilerplate.py manually like #[Update boilerplate for 2018](https://github.com/kubernetes/kubernetes/commit/bec420875ea5c64a9c4fa1269ada2a856b3d5ab7#diff-0f66228ea46785f57b8df9ca08b23f49), #[Update boilerplate.py to support 2017](https://github.com/kubernetes/kubernetes/commit/98534200bce7d2e039d6f127af828bd64dfe6086#diff-0f66228ea46785f57b8df9ca08b23f49), it should be auto checked.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes # [Update boilerplate for 2018](https://github.com/kubernetes/kubernetes/commit/bec420875ea5c64a9c4fa1269ada2a856b3d5ab7#diff-0f66228ea46785f57b8df9ca08b23f49) , #[Update boilerplate.py to support 2017](https://github.com/kubernetes/kubernetes/commit/98534200bce7d2e039d6f127af828bd64dfe6086#diff-0f66228ea46785f57b8df9ca08b23f49)

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: hack/boilerplate/boilerplate.py [hunk 1]
```diff
@@ -171,12 +172,23 @@ def get_files(extensions):
...
+    dates = dates[1:]
+    return '(' + dates + ')'
+
 def get_regexs():
     regexs = {}
     # Search for "YEAR" which exists in the boilerplate, but shouldn't in the real thing
     regexs["year"] = re.compile( 'YEAR' )
-    # dates can be 2014, 2015, 2016, 2017, or 2018; company holder names can be anything
-    regexs["date"] = re.compile( '(2014|2015|2016|2017|2018)' )
+    # get_dates return 2014, 2015, 2016, 2017, or 2018 until the current year as a regex like: "(2014|2015|2016|2017|2018)";
```
Comment (CaoShuFeng) [inline hack/boilerplate/boilerplate.py]: There is a trailing whitespace in this line.

**Comment by CaoShuFeng** [comment]
Seems good to me overall.
Thanks.

File: hack/boilerplate/boilerplate.py (hunk 1 above)
Comment (fisherxu) [reply hack/boilerplate/boilerplate.py]: thanks, have fixed, wondering how do you foung it  :-)

Comment (CaoShuFeng) [reply hack/boilerplate/boilerplate.py]: $ git am 60866.patch
应用：auto check the current year
.git/rebase-apply/patch:37: trailing whitespace.
    # get_dates return 2014, 2015, 2016, 2017, or 2018 until the current year as a regex like: "(2014|2015|2016|2017|2018)"; 
warning: 1 行有空白字符误用。

File: hack/boilerplate/boilerplate.py [hunk 2]
```diff
@@ -24,6 +24,7 @@
 import os
 import re
 import sys
+import datetime
```
Comment (CaoShuFeng) [inline hack/boilerplate/boilerplate.py]: It looks like these imports are in alphabetical order.

Comment (fisherxu) [reply hack/boilerplate/boilerplate.py]: Done.

File: hack/boilerplate/boilerplate.py [hunk 3]
```diff
@@ -171,12 +172,23 @@ def get_files(extensions):
...
 
+def get_dates():
+    dates = ''
+    years = datetime.datetime.now().year
+
+    for year in range(2014, years+1):
+        dates += '|' + str(year)
+
+    dates = dates[1:]
+    return '(' + dates + ')'
```
Comment (ixdy) [inline hack/boilerplate/boilerplate.py]: if you enjoy "python golf", you can combine a bunch of this together with the `str.join()` function and a list comprehension:
```python
return '(%s)' % '|'.join((str(y) for y in range(2014, years)))
```

(avoids the `dates[1:]` edge case, too)

**Review by ixdy — COMMENTED** [review]
/lgtm

(Though this fixes only half of the new year problem; the other problem is that our checked-in generated code includes the year and thus causes diffs in the new year. Some day we will fix that, too.)

**Comment by ixdy** [comment]
LGTM too. Thanks!

**Comment by cblecker** [comment]
As this only impacts the verify-boilerplate job, and will need to be picked to the 1.10 branch in the future anyways, I'm adding this to the milestone.

@jdumars any objections?

**Comment by dims** [comment]
/status approved-for-milestone

(on behalf of sig-release)

**Pull request merged**